pwiz_tools/Skyline/FileUI/PeptideSearch/TransitionSettingsControl.cs

```diff
@@ -349,11 +352,30 @@ public TransitionSettings GetTransitionSettings(Form parent)
                 settings.Instrument.MaxInclusions, settings.Instrument.MinTime, settings.Instrument.MaxTime);
             Helpers.AssignIfEquals(ref instrument, settings.Instrument);
 
-            TransitionLibraryPick pick = settings.Libraries.Pick != TransitionLibraryPick.none ? settings.Libraries.Pick : TransitionLibraryPick.all;
+            var pick = settings.Libraries.Pick != TransitionLibraryPick.none && DocumentSettings.PeptideSettings.Libraries.LibrarySpecs.Any(lib => !lib.IsDocumentLibrary)
+                ? settings.Libraries.Pick
+                : TransitionLibraryPick.filter;
+            if (CountFinder != null && pick == TransitionLibraryPick.filter)

```

Comment:
Comment here too. Even though I understand this one, it still deserves a quick comment.
// If IonRangeTo is a specific ion count, then the range restriction needs to be ignored in the
// presence of a spectral library for choosing ions.